libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/accounts/MapboxNavigationAccounts.kt

```diff
@@ -71,6 +73,11 @@ class MapboxNavigationAccounts private constructor() : SkuTokenProvider {
         } ?: resourceUrl
     }
 
+    /**
+     * Returns current SDK SKU token needed for ART.
+     */
+    override fun obtainSkuToken(): String = skuGenerator?.generateToken() ?: EMPTY_SKU
```

Comment:
Does this need to be `@Synchronized` 🤔 @kmadsen @abhishek1508

Reply:
if it does, would prefer the callback approach over blocking